coalib/output/ConsoleInteraction.py

```diff
@@ -922,27 +933,58 @@ def print_bears(bears,
     :param show_params:      True if the parameters and their description
                              should be shown.
     :param console_printer:  Object to print messages on the console.
+    :param args:             Args passed to coala command.
     """
     if not bears:
         console_printer.print('No bears to show. Did you forget to install '
                               'the `coala-bears` package? Try `pip3 install '
                               'coala-bears`.')
         return
 
+    results = []
+    if args and args.json:
+        from coalib.output.JSONEncoder import create_json_encoder
+        JSONEncoder = create_json_encoder(use_relpath=args.relpath)
+
     for bear, sections in sorted(bears.items(),
                                  key=lambda bear_tuple:
                                  bear_tuple[0].name.lower()):
-        show_bear(bear,
-                  show_description,
-                  show_params,
-                  console_printer)
+        if args and (args.json or args.format):
+            results.append(bear)
+        else:
+            show_bear(bear,
+                      show_description,
+                      show_params,
+                      console_printer)
+
+    if args:
+        if args.json:
+            json_output = {'bears': results}
+            import json
+            if args.output:
```

Comment:
what if output is provided for `format`?

Reply:
As per help section,
```
-o, --output          write results to the given file (must be called with
                        --json)
```

IMO, if we want this option to be used with other entrypoints in the similar context, we should file different issue.